Assets/Scripts/Entity/Player/Sanity/SanityController.cs

```diff
@@ -16,64 +14,260 @@ public enum StatType { Health, Saturation, Hydration };
+        private float healMultiplier;
+
+        [Tooltip("The duration of the heal bonus in seconds.")]
+        [SerializeField]
+        private float healBonusDuration;
+
+        [Tooltip("Stop any negative sanity changes caused by insufficient health, saturation, hydration when in range of a campfire.")]
+        [SerializeField]
+        private bool applyCampfireBonus;
+
+
+        [Header("More negative event ticks")]
+        [Tooltip("The time in seconds you have to be in combat to get a fight malus.")]
+        [SerializeField]
+        private float criticalFightTime;
+
+        [Tooltip("The malus for being in a fight for a certain time.")]
+        [SerializeField]
+        private float fightMalus;
+
+        [Tooltip("The malus for being in a fight with friendly NPCs for a certain time.")]
+        [SerializeField]
+        private float fightNonAggressivesMalus;
+
+        [Tooltip("The time you have to be out of combat (not attacking, not being hit) to be considered out of combat.")]
+        [SerializeField]
+        private float fightStopTime;
+
+        [Tooltip("The number of times to be hit to get a hit malus.")]
+        [SerializeField]
+        private int criticalHitCount;
+
+        [Tooltip("The malus for being hit a certain number of times. (Kind of a \"long fight\" malus.)")]
+        [SerializeField]
+        private float hitMalus;
+
+
         private PlayerState _playerState;
+        // for getting the target
+        private AttackLogic _attackLogic;
+
+        // depending on the player's needs build up to 1 or -1 for the next tick
+        private float _needsTick;
+        // for pausing any negative ticks caused by the player's needs
+        private bool _pauseTick;
+        // depending on events on the player build up a negative tick
+        private float _eventTick;
+        // whether the player is considered to be in combat
+        private bool _isFighting;
+        // the time the player has been in combat up until now
+        private float _fightTime; 
+        // (if you attack a hostile NPC but change targets to a friendly NPC just before criticalFightTime 
+        // is reached, you'll get the fightNonAgressivesTick)
+
+        // the time the player has stopped fighting while being in combat
+        private float _attackStopTimer;
+        // the number of hits performed on the player
+        private int _hitCounter;
+        // whether heal bonus is currently active
+        private bool _healBonus;
+        // a timer counting up to stop deactivate the heal bonus after a certain time
+        private float _healBonusStopTimer;
 
         private void Awake()
         {
             _playerState = GetComponent<PlayerState>();
+            _attackLogic = GetComponent<AttackLogic>();
         }
         
         private void OnEnable()
         {
             PlayerState.OnPlayerHealthUpdate += (int health) => sanityMath.InfluenceSanityByStat(StatType.Health, health);
```

Comment:
Lambda functions cannot be unsubscribed!

Reply:
Ahh ya I saw that as well, forgot to fix it